### ISSUE TITLE
Only works for files in current directory

### ISSUE THREAD

**Issue by EtiamNullam** [lukas-reineke/indent-blankline.nvim#52]
This plugin seems to only work on files in current directory. Looks like a problem with slashes or line ending characters on Windows.

Does not work for any of these paths:
```
.\Example.cs
Desktop\Example.cs
D:\Example.cs
..\Example.cs
~\Example.cs
```

Works fine only when file is opened directly like `nvim Example.cs`. Executing `:cd ..` after it is enough to make it stop working again.

OS: Windows 10
Neovim: v0.4.4

Minimal vimrc:
```viml
call plug#begin('~/.vim-plug')

    Plug 'lukas-reineke/indent-blankline.nvim'
    Plug 'Yggdroot/indentLine'
    Plug 'sheerun/vim-polyglot'

call plug#end()
```

## Some more info that might be useful according to troubleshooting in other issues

- Result of `verbose setlocal indentexpr?` (in typescript file):

```viml
indentexpr=GetTypescriptGraphQLIndent()
	Last set from ~\.vim-plug\vim-polyglot\after\indent\typescript.vim line 37
```

- This prints "hello" as expected:
```viml
:echon rpcrequest(g:indent_blankline_nvim_instance, 'nvim_eval', '"hello"')
```

- Nothing in `:messages` after running with this line in vimrc:
```viml
let g:indent_blankline_debug = v:true
```
...unless when its a VimL file:
```
indent-blankline encountered an error while finding matches: Vim:E492: Not an editor command: viminit.vim
```
...or when you open some help page (in this case simple `:help`):
```
indent-blankline encountered an error on refresh: Vim(call):E12: Command not allowed from exrc/vimrc in current dir or tag search
indent-blankline encountered an error while finding matches: Vim:E492: Not an editor command: eovimNeovimshare^@vim^Muntimedochelp.txt
```

**Comment by lukas-reineke** [comment]
Sorry I can't support windows for version 1.
I don't have access to a windows install, so I can't test anything.

If you update to Neovim 0.5 you can use the lua branch. That should work fine on windows. (also no need for `Yggdroot/indentLine` anymore, plus a lot of other improvements)


As I am only waiting for Neovim 0.5 to become stable before the lua rewrite will be the default, I am not really interested in big changes to the current version.
But if you can fix it yourself, feel free to open a PR.

**Comment by EtiamNullam** [comment]
That's sad but it was to be expected.

I have found out slashes disappear when they are sent via `rpcnotify` (meant for escaping characters maybe?):

https://github.com/lukas-reineke/indent-blankline.nvim/blob/207e001be64b39b72f8661e4a723d5bab698b5cd/autoload/indent_blankline.vim#L98

Looks like by enabling `shellslash` option (only for Windows) slashes will go through just fine (and no other text mangling so far). Some paths still refused to work but sending full path by `expand('%:p')` apparently fixed it.

This hack works for me so far:

autoload/**indent_blankline.vim**
- original:
```viml
    ...

    let l:file = expand('%')

    ...
```
- patched:
```viml
    ...

    set shellslash

    let l:file = expand('%:p')

    set noshellslash

    ...
```

I know its not perfect but maybe it will help someone. I never had a need to enable `shellslash` so I'm okay with keeping it disabled and enabling just for this single `expand`.

Can't wait for Neovim 0.5. That's another reason to try it but I need it to be stable.

**Comment by lukas-reineke** [comment]
Thanks for the work. If its just `shellslash` that fixes it, I am happy to merge it in if you make a PR.

If you make the PR, don't just do `set noshellslash`, users might have set it, we don't want to change their config.
You'll need to save the value it is currently `let l:shellslash = &shellslash`, then reset it back to that value after `set shellshlash=l:shellslash`.

**Comment by EtiamNullam** [comment]
I'm aware the previous hack will override user setting, that's why I called it a hack. I failed to provide anything better at that moment, I don't have much vimscript experience.

`:echo &shellslash` prints `0` (or `1`) and `:set shellslash = 0` will fail (`E474: Invalid argument`), so we can't use `let l:shellslash = &shellslash`.

I came up with this:

```viml
    ...

    let l:keep_shellslash_disabled = !&shellslash

    set shellslash

    let l:file = expand('%:p')

    if l:keep_shellslash_disabled
        set noshellslash
    endif

    ...

```

It seems to preserve what setting I have prior to refresh. I will give it a bit more testing and create a PR when possible.
